ios/Capacitor/Capacitor/Plugins/Haptics.swift

```diff
@@ -23,6 +23,25 @@ public class CAPHapticsPlugin: CAPPlugin {
       }
     }
   }
+
+  @objc public func notification(_ call: CAPPluginCall) {
+      DispatchQueue.main.async {
+          if let type = call.options["type"] as? String {
+              var notificationType = UINotificationFeedbackGenerator.FeedbackType.success
+              
+              if type == "WARNING" {
+                  notificationType = UINotificationFeedbackGenerator.FeedbackType.warning
+              } else if type == "ERROR" {
+                  notificationType = UINotificationFeedbackGenerator.FeedbackType.error
+              }
+              let generator = UINotificationFeedbackGenerator()
```

Comment:
Instead of having a duplicated generator, you can put it before the first if and use the same on both cases

Reply:
done